### ISSUE TITLE
uv_spawn_error: operation not supported on socket

### ISSUE THREAD

**Issue by Tulrath** [libuv/libuv#494]
After upgrading to electron-prebuilt 0.31.0, I started receiving the error "uv_spawn_error: operation not supported on socket" when trying to execute a uv_spawn() command.  This code functioned correctly before the electron-prebuilt update (PATHMAX is a #define to help support cross-platform, pid is set correctly higher in the script, and installerPath32 is getting set correctly in the GetInstallerPath() function).

```javascript
TCHAR installerPath32[PATHMAX];
	GetInstallerPath(installerPath32, false);
	char *args32[4];
	args32[0] = installerPath32;
	args32[1] = pid;
	args32[2] = "Wait";
	args32[3] = NULL; // libuv requires last arg to be a NULL

  uv_process_options_t options = {0};
	options.exit_cb = GameTrackerPlugin::OnExit;
	options.file = args32[0];
	options.args = args32;
  if(uv_default_loop())
  {
    if ((r = uv_spawn(uv_default_loop(), &child_req, &options))) {
      fprintf(stderr, "uv_spawn_error: %s\n", uv_strerror(r));
      return false;
    } else {
      fprintf(stdout, "Launched %s with pid %d\n", installerPath32, child_req.pid);
    }
  }
  else
  {
    fprintf(stderr, "uv_default_loop was invlid");
    return false;
  }

	return true;
}

```

The output I'm getting is:

```
uv_spawn_error: operation not supported on socket
```

Any help, or insights would be greatly appreciated :+1:

**Comment by bnoordhuis** [comment]
What version of libuv is that and what did you use before?

**Comment by Tulrath** [comment]
Sorry for being such a newbie, but could you tell me how to display the libuv version (Google failed me for this information)

**Comment by saghul** [comment]
@Tulrath http://docs.libuv.org/en/latest/version.html?highlight=version

**Comment by Tulrath** [comment]
So... I need to write and run a javascript program to display the libuv version?  Is that correct?

**Comment by cjihrig** [comment]
If you're using JavaScript (node.js), you should be able to just echo `process.versions`.

**Comment by Tulrath** [comment]
Looks like it's libuv 1.6.1

Here's how I got the information (not sure if this is the correct information?):

cat  ~/.electron-gyp/.node-gyp/0.31.0/deps/uv/include/uv-version.h

```
#define UV_VERSION_MAJOR 1
#define UV_VERSION_MINOR 6
#define UV_VERSION_PATCH 1
#define UV_VERSION_IS_RELEASE 1
#define UV_VERSION_SUFFIX ""
```

**Comment by bnoordhuis** [comment]
There have been quite a few fixes since v1.6.1 so my first suggestion would be to upgrade the embedded libuv and try again.

**Comment by saghul** [comment]
No follow up, closing. If you still see the failure with the latest libuv please leave a comment and we'll reopen the issue, thanks!
